src/americana.js

```diff
@@ -58,9 +58,7 @@ export const map = (window.map = new maplibregl.Map({
 
 CustomShields.loadCustomShields();
 
-map.on("styledata", function (event) {
-  ShieldDef.loadShields(map.style.imageManager.images);
-});
+ShieldDef.loadShields();
```

Comment:
It’s no longer necessary to defer compiling the shield definitions until the style requests a missing image, but maybe it could be deferred until the [style data begins loading](https://maplibre.org/maplibre-gl-js-docs/api/map/#map.event:styledataloading) instead.

Reply:
I find that if I wrap these loads, it breaks shieldtest.

Reply:
It probably has something to do with shieldtest only waiting for the `load` event:

https://github.com/ZeLonewolf/openstreetmap-americana/blob/067a48ff546ef482022cf062049247cd7f749134/src/shieldtest.js#L13

While it isn’t incorrect to build the shield definitions on page load, it’s suboptimal that shieldtest would be picky about when this happens. I don’t think shieldtest is actually making use of the map at all; it should just call `loadShields()` directly without importing americana.js or waiting for any map event.

Reply:
Actually it’s using the map to get the spritesheet to pass into `generateBlankGraphicsContext()`. I guess this is arguably necessary for covering the full shield rendering code path, but it’s mildly annoying that all the non-shield-related map code needs to come along for the ride.

Reply:
shieldtest.js separated from americana.js in 4a960e4.  Now we no longer have to take the layers along for the ride.

Reply:
I tried changing the load functions to:

```
map.on("styledataloading", function () {
  CustomShields.loadCustomShields();
  ShieldDef.loadShields();
});
```
Shields do **not** load when I do this, either in americana.js or shieldtest.

Reply:
If I change it to the "sourcedata" event, that works, presumably because that happens prior to style loading.  Does that meet the intent?  In any case, the shield definition loading should be a lightweight event so I'm not sure why it's important to defer it.

Reply:
I was thinking that it might be useful to defer it based on any user preferences we decide to vary the style by, but we could always do that explicitly rather than relying on GL JS’s event system. As long as this step is lightweight, we can leave it as is.

Reply:
I put these two calls back inline.  All they do is construct JSON objects so that seems safe.